modules/imgproc/src/opencl/separated_morph.cl

```diff
@@ -0,0 +1,44 @@
+
+// complile-time definitions
+// THRESHOLD, RADIUS, POW_RADIUS
+
+__constant sampler_t sampler = CLK_NORMALIZED_COORDS_FALSE |
+                               CLK_ADDRESS_CLAMP_TO_EDGE | CLK_FILTER_NEAREST;
+
+#define IS_SET(value) value[0] > 200
+
+#define WITHIN(value, iter) value + pown((float)(iter), 2) < POW_RADIUS
+
+__kernel void
+HorizontalFilter(__read_only image2d_t inputImage,
+                 __write_only image2d_t outputImage, unsigned int _radius) {
```

Comment:
OpenCV uses buffers as storage container. Images are not universal as we want (no support for multi-dimensional arrays, no support for all OpenCV border processing modes).

Reply:
cool, will adjust it once we have sorted the higher level stuff out